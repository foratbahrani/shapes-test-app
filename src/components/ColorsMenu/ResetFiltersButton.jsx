import React from 'react';
import { connect } from 'react-redux';
import { resetFilters } from '../../store/shapes/shapes.actions';
import Shape from '../Shape/Shape';
import './ColorsMenuButton.scss';

export const ResetFiltersButton = ({ color, resetFilters }) => {
	return (
		<button className={`colorsMenuButton`} id='button' onClick={() => resetFilters()}>
			<Shape key={color} shape='square' size='sm' />
		</button>
	);
};

const mapDispatchToProps = {
	resetFilters: resetFilters,
};

export default connect(null, mapDispatchToProps)(ResetFiltersButton);
