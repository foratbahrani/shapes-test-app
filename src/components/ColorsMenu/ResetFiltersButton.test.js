import { shallow } from 'enzyme';
import React from 'react';
import { ResetFiltersButton } from './ResetFiltersButton';

let wrapper;
let resetFilters = undefined;

beforeEach(() => {
	const color = 'red';
	const filteredColors = ['red'];
	resetFilters = jest.fn().mockReturnValue('');
	wrapper = shallow(
		<ResetFiltersButton
			color={color}
			filteredColors={filteredColors}
			resetFilters={resetFilters}
		/>
	);
});

it('matches snapshot', () => {
	expect(wrapper).toMatchSnapshot();
});

it('calls reset function', () => {
	expect.assertions(2);
	wrapper.find('[id="button"]').simulate('click');
	expect(resetFilters.mock.calls.length).toBe(1);
	wrapper.find('[id="button"]').simulate('click');
	expect(resetFilters.mock.calls.length).toBe(2);
});
