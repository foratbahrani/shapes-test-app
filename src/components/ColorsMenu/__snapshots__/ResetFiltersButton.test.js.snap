// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="colorsMenuButton"
  id="button"
  onClick={[Function]}
>
  <Shape
    key="red"
    shape="square"
    size="sm"
  />
</button>
`;
